js/src/forum/components/LogInButtons.js

```diff
@@ -20,6 +21,28 @@ export default class LogInButtons extends Component {
    * @public
    */
   items() {
-    return new ItemList();
+    const items = new ItemList();
+
+    const providers = app.forum.data.attributes.ssoProviders;
+
+    for (const provider in providers) {
+      const className = `Button LogInButton--${provider}`;
+      const path = `/auth/${provider}`;
+      var style = "";
+      style += providers[provider].buttonColor ? `background-color: ${providers[provider].buttonColor};` : '';
+      style += providers[provider].buttonTextColor ? `color: ${providers[provider].buttonTextColor}` : '';
+      items.add(provider,
+        <LogInButton
```

Comment:
I have not really seen how it looks like for extensions after this change, but I feel they loose some control over their login buttons. They are restricted to only the attributes hard-coded here but might want to customize others.

Reply:
That's true, I wanted to make it easier for extensions, and these are the settings that every auth extension uses so far. If extensions wanted to go beyond this, couldn't they just extend the LogIn Buttons itemlist?